server/verify_test.go

```diff
@@ -1243,6 +1264,19 @@ func TestVerifyAttestationWithTdx(t *testing.T) {
 			attest:  attestation2,
 			wantErr: "failed to verify memory encryption technology: could not interpret Root CA certificate DER bytes: x509: invalid RDNSequence: invalid attribute value",
 		},
+		{
+			name: "Wrong TEE Nonce",
+			opts: VerifyOpts{
+				Nonce:      nonce,
+				TrustedAKs: []crypto.PublicKey{ak.PublicKey()},
+				TEEOpts: &VerifyTdxOpts{
+					Validation:   TdxDefaultValidateOpts([]byte("badNonce")),
+					Verification: tv.DefaultOptions(),
+				},
+			},
+			attest:  attestation,
+			wantErr: "quote field REPORT_DATA",
+		},
 	}
```

Comment:
In general, these tests are difficult to read due to the large structs getting created in the test table. Instead of creating them here, let's populate these tables with the actual variables we care about (`VerifyOpts.nonce`, the TEE nonce, `attestation`, `wantErr`, etc.). Then create the VerifyOpts object in the `t.Run` block.

Reply:
Done